File: src/calendar-backend/Calendar.Mongo.Db/Models/CalendarEvent.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson.Serialization.Options;

namespace Calendar.Mongo.Db.Models
{
    public class CalendarEvent
    {
        [BsonId]
        [BsonRequired]
        public ObjectId Id { get; set; }

        [BsonRequired]
        public string? Location { get; set; }

        public string? Description { get; set; }

        [BsonRequired]
        [BsonRepresentation(BsonType.String)]
        public DateTimeOffset Start { get; set; }

        [BsonRequired]
        public TimeSpan Duration { get; set; }

        public ObjectId? SeriesId { get; set; }

        public EventRepeat Repeat { get; set; }

        [BsonRepresentation(BsonType.String)]
        public DateTimeOffset? StartSeries { get; set; }

        [BsonRepresentation(BsonType.String)]
        public DateTimeOffset? EndSeries { get; set; }

        // do not ToUniversalTime() because only used in api so not need!
        [BsonRequired]
        [BsonRepresentation(BsonType.String)]
        public DateTimeOffset CreatedDate { get; set; }
        // do not ToUniversalTime() because only used in api so not need!
        [BsonRequired]
        [BsonRepresentation(BsonType.String)]
        public DateTimeOffset? LastUpdateDate { get; set; }

        public List<Instructor>? Instructors { get; set; }

        [BsonRequired]
        public string? LectureId { get; set; }


        [BsonRequired]
        public string? CalendarId { get; set; }

        public CalendarEvent() { }

    }
}

File: src/calendar-client/src/models/calendarEvent.ts
import { Calendar } from './calendar'
import { Lecture } from './lecture'
import { Moment } from 'moment/moment'
import { Instructor } from './instructor'

export interface CalendarEventBase {
    location: string
    description?: string
    start: Moment
    end: Moment
    instructors: Instructor[]
    repeat: number
    calendarId: string
}

export interface CalendarEvent extends CalendarEventBase {
    id: string
    endSeries?: string | undefined
    createdDate?: Moment
    lastUpdateDate?: Moment
    calendar: Calendar
    lecture: Lecture
    seriesId: string
    startSeries?: Moment
}

export interface CreateCalendarEvent extends CalendarEventBase {
    endSeries?: Moment | undefined
    lectureId: string
    calendarId: string
}

export interface UpdateCalendarEvent extends CalendarEventBase {
    id?: string
    calendarId: string
    seriesId?: string
    lectureId: string
    endSeries: Moment
    // this dates only necessary for api to check the last edit
    lastUpdateDate?: Moment
    createdDate?: Moment
    instructors: Instructor[]
}

export interface UpdateCalendarEventSeries extends CalendarEventBase {
    calendarId: string
    seriesId: string
    endSeries: Moment
    lectureId: string
    lastUpdateDate?: Moment
    createdDate?: Moment
}
